Goal: Fix chage display when the last change field is set to 0.
      This is consistent with PAM.

Status wrt upstream: It should be forwarded to upstream.

Index: shadow-cvs/src/chage.c
===================================================================
--- shadow-cvs.orig/src/chage.c	2005-10-06 02:48:24.000000000 +0300
+++ shadow-cvs/src/chage.c	2005-10-06 02:48:44.000000000 +0300
@@ -234,8 +234,10 @@
 	 * was last modified. The date is the number of days since 1/1/1970.
 	 */
 	printf (_("Last password change\t\t\t\t\t: "));
-	if (lastday <= 0) {
+	if (lastday < 0) {
 		printf (_("never\n"));
+	} else if (lastday == 0) {
+		printf (_("password must be changed"));
 	} else {
 		changed = lastday * SCALE;
 		print_date (changed);
